src/coreclr/nativeaot/Common/src/Internal/Runtime/CompilerHelpers/StartupCodeHelpers.cs

```diff
@@ -262,9 +262,44 @@ private static unsafe void RehydrateData(IntPtr dehydratedData, int length)
                 switch (command)
                 {
                     case DehydratedDataCommand.Copy:
-                        // TODO: can we do any kind of memcpy here?
-                        for (; payload > 0; payload--)
-                            *pDest++ = *pCurrent++;
+                        Debug.Assert(payload != 0);
+                        if (payload < 4)
```

Comment:
Does it regress if we use:

```c#
// > 64 hits p/invoke: https://github.com/dotnet/runtime/blob/5f51269/src/libraries/System.Private.CoreLib/src/System/Buffer.cs#L145
if (payload <= 64)
{
    Buffer.MemoryCopy(pDest, pCurrent, payload, payload);
}
else // existing code to avoid p/invoke - line 289 onwards
```

Reply:
It won't compile when this file is included from Test.CoreLib.csproj.

MemoryCopy is also so big it won't inline and we run this logic this tens of thousands of times in this tight loop (11k just for hello world). Inlining it is preferable.